addons/website_sale/controllers/main.py

```diff
@@ -772,15 +772,21 @@ def cart_update(
         if product_custom_attribute_values:
             product_custom_attribute_values = json_scriptsafe.loads(product_custom_attribute_values)
 
-        if no_variant_attribute_values:
-            no_variant_attribute_values = json_scriptsafe.loads(no_variant_attribute_values)
+        # old API, will be dropped soon with product configurator refactorings
+        no_variant_attribute_values = kwargs.pop('no_variant_attribute_values', None)
+        if no_variant_attribute_values and no_variant_attribute_value_ids is None:
+            no_variants_attribute_values_data = json_scriptsafe.loads(no_variant_attribute_values)
+            no_variant_attribute_value_ids = [
+                int(ptav_data['value'])
+                for ptav_data in no_variants_attribute_values_data
+            ]
```

Comment:
It can be on one line maybe
```suggestion
            no_variant_attribute_value_ids = [
                int(ptav_data['value']) for ptav_data in no_variants_attribute_values_data
            ]
```